### PR TITLE
#22 Create Category-Items-Screen

### PR DESCRIPTION
**Pull request by ZeyadSultan**
issue link: https://banquemisrmentorship.atlassian.net/browse/ES-22
issue no.: 22

### CODE REVIEW

File: e-store/e-store/ScenesScreen/CategoryItems/Model/CategoryItem.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//  Item.swift
+//  e-store
+//
+//  Created by Zeyad on 11/04/2023.
+//
+
+import Foundation
+
+class CategoryItem {
+    var img:ImageAsset
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/Model/CategoryItem.swift:11]: Modify it to image url ---> String

File: e-store/e-store/ScenesScreen/CategoryItems/Model/CategoryItem.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    var name: String
+    var price: Int
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/Model/CategoryItem.swift:13]: modify it to double

File: e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+     required init?(coder: NSCoder) {
+         fatalError("init(coder:) has not been implemented")
+     }
+
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        itemsCollectionView.layer.cornerRadius = 10
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.swift:32]: plz create new func like setupCollectionView ,

File: e-store/e-store/ScenesScreen/CategoryItems/Model/CategoryItem.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
+//
+//  Item.swift
+//  e-store
+//
+//  Created by Zeyad on 11/04/2023.
+//
+
+import Foundation
+import UIKit
+import Kingfisher
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/Model/CategoryItem.swift:10]: why import Kingfisher

File: e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.xib [hunk 5]
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="21507" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.xib:2]: plz handle nav bar from view controller

File: e-store/e-store/ScenesScreen/CategoryItems/Cell/ItemsCollectionViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    @IBOutlet private weak var itemPrice: UILabel!
+    @IBOutlet private weak var itemRating: UILabel!
+    @IBOutlet private weak var itemReviews: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
+    
+    public func setupOutlets(product: CategoryItem) {
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/Cell/ItemsCollectionViewCell.swift:24]:
```suggestion
    public func bindOutlets(product: CategoryItem) {
```

File: e-store/e-store/ScenesScreen/CategoryItems/Cell/ItemsCollectionViewCell.xib [hunk 7]
```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/Cell/ItemsCollectionViewCell.xib:1]: add trailing on item image 
remove the constraint between item on the same stack view

File: e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.xib (hunk 5 above)
Comment (Mahmoud-Bassuni) [reply e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.xib:2]: handle the title of category name from nav bar

**Review by Mahmoud-Bassuni — COMMENTED** [review]
thanks dear 🌹

File: e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.swift [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        let backArrow = UIBarButtonItem(image: UIImage(systemName: "chevron.backward"), style: .plain, target: self, action: #selector(backArrowTapped))
+         
+        backArrow.tintColor = UIColor.navyBlackColor
+        navigationItem.leftBarButtonItem = backArrow
+        
+        let cart = UIBarButtonItem(image: UIImage(systemName: "cart"), style: .plain, target: self, action: #selector(cartTapped))
+        cart.tintColor = UIColor.navyBlackColor
+        navigationItem.rightBarButtonItem = cart
+        
+//        let subtitleLabel = UILabel()
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/CategoryItems/ViewController/CategoryItemsViewController.swift:51]: plz remove those comments

**Review by Mahmoud-Bassuni — DISMISSED** [review]

**Review by hassanSayed0 — APPROVED** [review]

**Review by AhmedAtef277 — APPROVED** [review]

**Pull request merged**